### PR TITLE
Enable CIDR netblocks in `from' fields

### PR DESCRIPTION
**Pull request by xdgc**
Some tests accompany.

### CODE REVIEW

File: authprogs/authprogs.py [hunk 1]
```diff
@@ -188,14 +189,24 @@ def check_client_ip(self, rule):
...
+                return None
+        allow_from = [
+            ipnet(x.decode())
+            for x in allow_from
+        ]
+        allow_from = filter(lambda x: x, allow_from)
+        client_ip = ipaddress.ip_address(self.get_client_ip().decode())
+
+        for allow in allow_from:
+            if allow == u'*' or client_ip in allow:
```
Comment (bcl) [inline authprogs/authprogs.py]: This wildcard doesn't work, it is clobbered by the ipnet function. 0.0.0.0/0 does work though :)

File: authprogs/tests/test_authprogs.py [hunk 2]
```diff
@@ -133,6 +134,12 @@ def test_client_ip(self):
         ap = getap('0.0.0.1', 'MULTIPLE_TWO')
         self.assertRaises(authprogs.CommandRejected, ap.find_match)
 
+    def test_client_ip_cidr(self):
+        """Test that a single bare command works."""
```
Comment (bcl) [inline authprogs/tests/test_authprogs.py]: The comment needs to be updated, something like 'Test a cidr network matches'

File: authprogs/tests/test_authprogs.py [hunk 3]
```diff
@@ -133,6 +134,12 @@ def test_client_ip(self):
...
+        ap = getap('10.11.12.13', 'CIDR')
+        self.assertEqual(ap.find_match(),
+                {'command': ['CIDR']})
+
```
Comment (bcl) [inline authprogs/tests/test_authprogs.py]: This should also include a failure check, like:

     ap = getap('0.0.0.1', 'CIDR')
     self.assertRaises(authprogs.CommandRejected, ap.find_match)

File: authprogs/authprogs.py (hunk 1 above)
Comment (xdgc) [reply authprogs/authprogs.py]: O rite. Iteration detritus. I wanted a shorthand for "all" but the mapping should be in ipnet, not here.

File: authprogs/tests/test_authprogs.py (hunk 2 above)
Comment (xdgc) [reply authprogs/tests/test_authprogs.py]: Good eye

File: authprogs/tests/test_authprogs.py (hunk 3 above)
Comment (xdgc) [reply authprogs/tests/test_authprogs.py]: Fair

**Comment by xdgc** [comment]
Talked with @daethnir and he's interested in resolving some python2 encoding issues by just ditching python2, which works for me. So not sure how applicable this all is, but I pushed fixes per @bcl comments for whatever value that has.

**Pull request merged**